genesyscloud/outbound_ruleset/resource_genesyscloud_outbound_ruleset.go

```diff
@@ -131,3 +144,59 @@ func deleteOutboundRuleset(ctx context.Context, d *schema.ResourceData, meta int
 		return retry.RetryableError(fmt.Errorf("Outbound Ruleset %s still exists", d.Id()))
 	})
 }
+
+// filterOutboundRuleset filters rulesets by removing the ones that are referencing skills that no longer exist in GC
+func filterOutboundRulesets(ruleSets []platformclientv2.Ruleset, skillMap resourceExporter.ResourceIDMetaMap) ([]platformclientv2.Ruleset, diag.Diagnostics) {
+	var filteredRuleSets []platformclientv2.Ruleset
+	log.Printf("Filtering outbound rule sets")
+
+RuleSetLoop:
+	for _, ruleSet := range ruleSets {
```

Comment:
I would highly recommend you break this function up.  In Go, if use see multi-level for and if statements, it is usually telling you that your code is too complex.  I would at least take that inner set of if and for loops and refactor it out to a single method call.  It will make it easier to write unit tests and prove that your logic is solid.